### DISCUSSION TITLE
Prompt drone for easy use(and a little question about the code)

### DISCUSSION THREAD

**Discussion by morningsunshine0401** [microsoft/PromptCraft-Robotics#17] [LLM + Aerial robots]
Hi, I am a student currently using the provided code and testing it on my drone for a personal project. I am using it to perform and succeeded simple movements such as "go up 40cm then land" or "fly aroung making a square". So for my case, will it be better if I get rid of the appending part from 'def ask'? Because I won't be needing past conversations from the previous questions. I think appending the chat_history is making it take more time and use too many tokens while performing.

+ I am planning to do a project that uses two different chatgpt api models to perform a faster task

Thanks for the awesome work by the way!

**Comment by Xj1567589354** [comment]
"So for my case, will it be better if I get rid of the appending part from 'def ask'? Because I won't be needing past conversations from the previous questions. ", what's the problem？

**Reply by Xj1567589354** [reply]
你可以做这个实验，

**Reply by Xj1567589354** [reply]
如果不传入历史记录，每次调用API都会像一个全新的对话一样，模型无法参考之前的指令和互动，也就无法进行连续的任务执行和对话。
